src/block_cache/block_io.c

```diff
@@ -99,6 +101,12 @@ __wt_blkcache_read(WT_SESSION_IMPL *session, WT_ITEM *buf, const uint8_t *addr,
             blkcache_found = true;
             ip->data = blkcache_item->data;
             ip->size = blkcache_item->data_size;
+            if (block_meta != NULL) {
+                if (blkcache_item->block_meta == NULL)
+                    memset(block_meta, 0, sizeof(*block_meta));
+                else
+                    memcpy(block_meta, &blkcache_item->block_meta, sizeof(*block_meta));
```

Comment:
If possible, use WT_CLEAR instead of memset, and struct assignment instead of memcpy.